metricflow/model/validations/measures.py

```diff
@@ -415,6 +415,63 @@ def validate_model(model: UserConfiguredModel) -> List[ValidationIssueType]:  #
         return issues
 
 
+class PercentileAggregationRule(ModelValidationRule):
+    """Checks that only PERCENTILE measures have agg_params and valid percentile value provided."""
+
+    @staticmethod
+    @validate_safely(
+        whats_being_done="running model validation ensuring the agg_params.percentile value exist for measures with percentile aggregation"
+    )
+    def validate_model(model: UserConfiguredModel) -> List[ValidationIssueType]:  # noqa: D
+        issues: List[ValidationIssueType] = []
+
+        for data_source in model.data_sources:
```

Comment:
We generally prefer to have a separate function for the data source level checks with the `@validate_safely` decorator, as it helps us understand exactly what's happening in the validation stack.

Reply:
every other measure validation (except MeasureConstraintAliasesRule) seems to do data source level checks in the main validate_model function without having a separate function, would you like me to separate those?

And would the main validate_model just look like: 

`def validate_model(model: UserConfiguredModel) -> List[ValidationIssueType]:  # noqa: D
        issues: List[ValidationIssueType] = []

        for data_source in model.data_sources:
            for measure in data_source.measures:
                context = DataSourceElementContext(
                    file_context=FileContext.from_metadata(metadata=data_source.metadata),
                    data_source_element=DataSourceElementReference(
                        data_source_name=data_source.name, element_name=measure.name
                    ),
                    element_type=DataSourceElementType.MEASURE,
                )
                issues.append(PercentileAggregationRule._validate_percentile_measure(measure, context))

        return issues`